rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -1272,6 +1272,39 @@ class DebruijnInterpreter[M[_]: Sync: Parallel: _cost](
       } yield result
   }
 
+  private[this] val split: Method = new Method() {
+
+    def split(baseExpr: Expr, sep: Expr): M[Expr] =
+      (baseExpr.exprInstance, sep.exprInstance) match {
+        case (GString(string), GString(sepStr)) =>
+          val split = string.split(sepStr)
+          charge[M](toListCost(split.size) + splitCost(string, sepStr, split.size)) >>
```

Comment:
@hilltracer Please verify if cost is correctly calculated?

Reply:
For split string with size 14:
“qwe1 qwe2 qwe3”.split(“ “) = RhoList(qwe1, qwe2, qwe3)

For me: 
size = 
14 - traverse each symbols of input string and compare with SepString
+
12 - Character-by-character copy of symbols from (qwe1qwe2qwe3) in array
+
toListCost(3) (now it’s equal 3, I think it’s wrong but it are used in many places - so doesn’t matter)

Conclusion:
```Cost = charge[M](toListCost(split.size) + splitCost(string, sepStr, split.size))```
where:  
```scala
def splitCost(str: String, sep: String, found: Int): Cost =
    Cost(2*str.getBytes.size - found * sep.length, “split”)
```
I think it’s a matter of taste.  You can ignore it, because yes...  it is probably possible to somehow optimize this and combine the traverse and copy operations, but then how to take into account the comparison with sep?

Reply:
If this is a matter of taste, then let's go with what's already coded? @tgrospic